### PR TITLE
Fix missing Python parameter type defaults

### PR DESCRIPTION
**Pull request by kieran-ryan**
### 🤔 What's changed?

The project documentation outlines for [Customer Parameter Types](prefer_for_regexp_match) that `use_for_snippets` "Defaults to `true`"  and that `prefer_for_regexp_match` "Defaults to `false`".

Although this aligns with the documented [Python example](prefer_for_regexp_match), its implementation is missing default values for those keyword arguments:

```python
class ParameterType:
    def __init__(
        self,
        name,
        regexp,
        type,
        transformer,
        use_for_snippets,
        prefer_for_regexp_match,
    ):
```

With this PR:

1. The appropriate defaults have been added to the `ParameterType` class. 
2. Two versions of Python type hints in the module defining that class have been reduced to one e.g. `typing.List` and `list` (through`__future__.annotations`).

### ⚡️ What's your motivation? 

- Fix a bug where default keyword arguments were not set for customer parameter types in Python
- Remove invalid documentation that alternatives can not be escaped

### 🏷️ What kind of change is this?

- :bug: Bug fix (non-breaking change which fixes a defect)

### ♻️ Anything particular you want feedback on?

Whether small correction to alternatives in README.md is appropriate.

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

### CODE REVIEW

**Comment by kieran-ryan** [comment]
Closes #142, providing a valid example of Cucumber Expressions with a Python framework, Behave.

File: README.md [hunk 1]
```diff
@@ -95,6 +95,8 @@ a parameter type.
 | `useForSnippets` / `use_for_snippets` | Defaults to `true`. That means this parameter type will be used to generate snippets for undefined steps. If the `regexp` frequently matches text you don't intend to be used as arguments, disable its use for snippets with `false`. |
 | `preferForRegexpMatch` / `prefer_for_regexp_match` | Defaults to `false`. Set to `true` if you have step definitions that use regular expressions, and you want this parameter type to take precedence over others during a match. |
 
+*Note: Currently the parameter name cannot be customized, so the custom parameters can only be used with the type name, e.g. `{Color}`.*
```
Comment (mpkorstanje) [inline README.md:98]: This doesn't sound accurate. I know it does work in Cucumber-JVM. Perhaps this should be a bug report for a specific Cucumber implementation?

File: README.md [hunk 2]
```diff
@@ -167,8 +169,6 @@ ParameterType(
 )
 ```
 
-*Note: Currently the parameter name cannot be customized, so the custom parameters can only be used with the type name, e.g. `{Color}`.*
```
Comment (mpkorstanje) [inline README.md:170]: This is presumably only for Python. But I'd leave it out of scope in this PR.

File: python/README.md [hunk 3]
```diff
@@ -1,38 +1,202 @@
...
 
-## Tests
+__Cucumber Expressions__ is available via [PyPI](https://pypi.org/project/cucumber-expressions/):
 
-The test suite uses `pytest` as its testing Framework.
+```console
+pip install cucumber-expressions
+```
+
+## Usage
```
Comment (mpkorstanje) [inline python/README.md:18]: Duplicating this from the root readme isn't great.

File: python/README.md [hunk 4]
```diff
@@ -1,38 +1,202 @@
...
+'^I have ((?:-?\\d+)|(?:\\d+)) cucumbers in my belly$'
+>>> match = expression.match("I have 42 cucumbers in my belly")
+>>> match
+[<cucumber_expressions.argument.Argument object at 0x10dcf7650>]
+>>> argument = match[0]
+>>> argument.value
+42
+```
+
+### Using Cucumber Expressions with Behave
```
Comment (mpkorstanje) [inline python/README.md:37]: This information is likely to go out of date without us noticing. It would be better to link directly to the behave docs.

Comment (mpkorstanje) [reply python/README.md:37]: Ultimately end users are not expected to ever see this package. I would expect Behave to wrap this API with its own.

So this documentation can be focused purely on using this API and this API alone.

**Review by mpkorstanje — CHANGES_REQUESTED** [review]
Overall this PR makes more changes then it says in the title. Fixing the defaults would have been an automatic merge, the other stuff needs some consideration.

File: python/README.md (hunk 4 above)
Comment (mpkorstanje) [reply python/README.md:37]: This is also true for the other examples btw. But I think those are linked from the docs.cucumber.io site. 

This isn't great either but at least that is within the same project.

File: README.md (hunk 1 above)
Comment (kieran-ryan) [reply README.md]: Moved to #277

File: README.md (hunk 2 above)
Comment (kieran-ryan) [reply README.md]: Precedes Python implementation, was associated with .NET example. Moved to #277

File: python/README.md (hunk 4 above)
Comment (kieran-ryan) [reply python/README.md]: That's fair - removed from PR. There's a slight gap at present between capturing this info between Cucumber and Behave docs - but hopefully Behave will add official native support at some point.

**Comment by kieran-ryan** [comment]
Appreciate the considered feedback @mpkorstanje - refined PR scope accordingly

**Pull request merged**